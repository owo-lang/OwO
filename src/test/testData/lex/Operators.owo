1+1=2
infixl 2 +
instance
  ListMonoid : Monoid List
  mempty ListMonoid = []
  (<>)   ListMonoid a b = a ++ b
concatM : {| Monoid m |} -> [m a] -> m a
concatM []       = mempty
concatM (a : as) = a <> concatM as
a<|>b<~>5
c>>=d.d
