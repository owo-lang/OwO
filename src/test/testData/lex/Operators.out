IntegerToken 1 (0 1 1) (1 1 2)
OperatorToken "+" (1 1 2) (2 1 3)
IntegerToken 1 (2 1 3) (3 1 4)
EqualToken (3 1 4) (4 1 5)
IntegerToken 2 (4 1 5) (5 1 6)
InfixLToken (6 2 1) (12 2 7)
IntegerToken 2 (13 2 8) (14 2 9)
OperatorToken "+" (15 2 10) (16 2 11)
InstanceToken (17 3 1) (25 3 9)
BraceLToken (28 4 3) (28 4 3)
IdentifierToken "ListMonoid" (28 4 3) (38 4 13)
ColonToken (39 4 14) (40 4 15)
IdentifierToken "Monoid" (41 4 16) (47 4 22)
IdentifierToken "List" (48 4 23) (52 4 27)
SemicolonToken (55 5 3) (55 5 3)
IdentifierToken "mempty" (55 5 3) (61 5 9)
IdentifierToken "ListMonoid" (62 5 10) (72 5 20)
EqualToken (73 5 21) (74 5 22)
BracketLToken (75 5 23) (76 5 24)
BracketRToken (76 5 24) (77 5 25)
SemicolonToken (80 6 3) (80 6 3)
ParenthesisLToken (80 6 3) (81 6 4)
OperatorToken "<>" (81 6 4) (83 6 6)
ParenthesisRToken (83 6 6) (84 6 7)
IdentifierToken "ListMonoid" (87 6 10) (97 6 20)
IdentifierToken "a" (98 6 21) (99 6 22)
IdentifierToken "b" (100 6 23) (101 6 24)
EqualToken (102 6 25) (103 6 26)
IdentifierToken "a" (104 6 27) (105 6 28)
OperatorToken "++" (106 6 29) (108 6 31)
IdentifierToken "b" (109 6 32) (110 6 33)
BraceRToken (111 7 1) (111 7 1)
IdentifierToken "concatM" (111 7 1) (118 7 8)
ColonToken (119 7 9) (120 7 10)
InstanceArgumentLToken (121 7 11) (123 7 13)
IdentifierToken "Monoid" (124 7 14) (130 7 20)
IdentifierToken "m" (131 7 21) (132 7 22)
InstanceArgumentRToken (133 7 23) (135 7 25)
RightArrowToken (136 7 26) (138 7 28)
BracketLToken (139 7 29) (140 7 30)
IdentifierToken "m" (140 7 30) (141 7 31)
IdentifierToken "a" (142 7 32) (143 7 33)
BracketRToken (143 7 33) (144 7 34)
RightArrowToken (145 7 35) (147 7 37)
IdentifierToken "m" (148 7 38) (149 7 39)
IdentifierToken "a" (150 7 40) (151 7 41)
IdentifierToken "concatM" (152 8 1) (159 8 8)
BracketLToken (160 8 9) (161 8 10)
BracketRToken (161 8 10) (162 8 11)
EqualToken (169 8 18) (170 8 19)
IdentifierToken "mempty" (171 8 20) (177 8 26)
IdentifierToken "concatM" (178 9 1) (185 9 8)
ParenthesisLToken (186 9 9) (187 9 10)
IdentifierToken "a" (187 9 10) (188 9 11)
ColonToken (189 9 12) (190 9 13)
IdentifierToken "as" (191 9 14) (193 9 16)
ParenthesisRToken (193 9 16) (194 9 17)
EqualToken (195 9 18) (196 9 19)
IdentifierToken "a" (197 9 20) (198 9 21)
OperatorToken "<>" (199 9 22) (201 9 24)
IdentifierToken "concatM" (202 9 25) (209 9 32)
IdentifierToken "as" (210 9 33) (212 9 35)
IdentifierToken "a" (213 10 1) (214 10 2)
OperatorToken "<|>" (214 10 2) (217 10 5)
IdentifierToken "b" (217 10 5) (218 10 6)
OperatorToken "<~>" (218 10 6) (221 10 9)
IntegerToken 5 (221 10 9) (222 10 10)
IdentifierToken "c" (223 11 1) (224 11 2)
OperatorToken ">>=" (224 11 2) (227 11 5)
IdentifierToken "d" (227 11 5) (228 11 6)
DotToken (228 11 6) (229 11 7)
IdentifierToken "d" (229 11 7) (230 11 8)
